### PR TITLE
refactor: refactoring query handling layer

### PR DESCRIPTION
**Pull request by himank**

### CODE REVIEW

File: store/kv/fdb.go [hunk 1]
```diff
@@ -318,7 +319,7 @@ func (t *ftx) Insert(_ context.Context, table string, key Key, data []byte) erro
 		return err
 	}
 	if vv != nil {
-		return os.ErrExist
+		return api.Errorf(codes.AlreadyExists, "duplicate key value, violates unique primary key constraint")
```
Comment (efirs) [inline store/kv/fdb.go:322]: I would not push dependency on API package that deep.

Comment (efirs) [reply store/kv/fdb.go:322]: Can we just have kv.ErrAlreadyExists?

**Review by efirs — APPROVED** [review]

File: store/kv/fdb.go (hunk 1 above)
Comment (himank) [reply store/kv/fdb.go:322]: This is not permanent here. I think we need to define errors for kv package and then translate in the server package. Just keeping it as a holder.

Comment (himank) [reply store/kv/fdb.go:322]: I'll also add a ToDo for this in follow up diff.

**Pull request merged**